### PR TITLE
feat: parser

### PR DESCRIPTION
**Pull request by AlexeiMet**

### CODE REVIEW

**Comment by AlexeiMet** [comment]
Reworked

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (ghost) [inline .vscode/settings.json:1]: Please include .vscode as well as other config files not related to the actual project into .gitignore

File: src/modules/parser/parser.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+            }
+        });
+    }
+
+    readFile(filepath) {
+        return new Promise((res, rej) => {
+            if (this.sync) {
+                fs.readFile(filepath, (err, data) => err ? rej(err) : res(data));
+            } else {
+                const data = fs.readFileSync(filepath);
```
Comment (ghost) [inline src/modules/parser/parser.js]: Here you still return a promise which will be moved out of the execution stack and handled in the event queue which is not truly 'sync' execution.

File: src/modules/parser/parser.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            }
+        });
+    }
+
+    readdir(dirpath) {
+        return new Promise((res, rej) => {
+            if (this.sync) {
+                fs.readdir(dirpath, (err, items) => err ? rej(err) : res(items));
+            } else {
+                const items = fs.readdirSync(dirpath);
```
Comment (ghost) [inline src/modules/parser/parser.js]: Here you still return a promise which will be moved out of the execution stack and handled in the event queue which is not truly 'sync' execution.

**Pull request merged**